### ISSUE TITLE
Version 3.1.3 Warning: Redis::pipeline(): Already in pipeline mode

### ISSUE THREAD

**Issue by valdislav** [phpredis/phpredis#1211]
Hi, today library was updated and we got this error. 

### Expected behaviour
Not throw warnings.
### Actual behaviour
Warning: Redis::pipeline(): Already in pipeline mode on every request. 
### I'm seeing this behaviour on
- OS: RHEL 6.9
- Redis: redis-2.8.17-2.el6.x86_64
- PHP: 7.0.21-1
- phpredis: php-pecl-redis-3.1.3-1.el6.remi.7.0.x86_64

### Steps to reproduce, backtrace or example script
In the middle php library olinmollenhour/cache-backend-redis: 1.9 used
Even simple session save handling fails
 
### I've checked
- [x] There is no similar issue from other users
- [x] Issue isn't fixed in `develop` branch

**Comment by michael-grunder** [comment]
I'll have to check but I think this is a new warning thrown if you try to enter pipeline mode when already in pipeline mode.

**Comment by remicollet** [comment]
@valdislav FYI, redis 3.1.3 have been removed from my repository, I recommend you downgrade to 3.1.2, waiting for some update.

**Comment by siliconalchemy** [comment]
+1 this has just killed production websites for us.  Downgrading to 3.1.2 fixed it.

**Comment by yatsukhnenko** [comment]
@michael-grunder I've specially added this warning to help users find and fix double `pipeline` calls because in previous releases this was dangerous.
As for me it is easier to fix this issue by adding check in php code instead of fixing in extension and releasing new version but I don't mind to remove this warning if it will affect many users.

**Comment by FadelChafai** [comment]
You can try to reinstall an older version of pecl/redis ( i had the same issue with Mag 1 )
pecl uninstall redis
pecl install redis-3.1.2

**Comment by DerekMarcinyshyn** [comment]
downgrading to redis-3.1.2 fixed it for me.
Magento EE 1.14.x 
last night upgraded from php 5.6.30 to php 5.6.31 which seemed to trigger the pipeline warning

**Comment by michael-grunder** [comment]
From my perspective I think that issuing a warning here is more _correct_ as entering pipeline mode twice (or multiple times) could indicate that the PHP client code has lost track of the object state. 

That being said, since this is new behavior introduced in 3.1.3 and is causing people pain it's probably best to go ahead and remove it.  There is nothing inherently dangerous about treating a duplicate call to `pipeline` as a NO OP.

I'll reference this issue with the commit.

**Comment by valugi** [comment]
Happens also in version 3.2.9.

**Comment by MikeParkin** [comment]
This is causing issues in Magento 1.x (warnings) and exceptions in Magento 2.x - lots of users will be affected.

**Comment by bkibilko** [comment]
The same problem exists on:
Magento 1.14.2.4
PHP 7.0.21
phpredis 3.1.3
redis 3.2.4
`Warning: Redis::pipeline(): Already in pipeline mode  in /var/www/magento/lib/Credis/Client.php on line 931`

**Comment by HOSTED-POWER** [comment]
Having same issue on Magento 2.x, redis 3.1.2 is working fine.

**Comment by michael-grunder** [comment]
If those affected by this would try with the latest version of the develop branch that would be great.

**Comment by nilands55s** [comment]
Ok for me this is not a Redis 3.1. or a php 5.6.31 version problem as my Magento 1.9.3.4 is on Ubuntu-Server 14.04 LTS. Where there is no upgrade beyond PHP 5.5.9 and Redis-server is version 2:2.8.4-2, SO I am not any where near the versions mentioned above and have the same issue.
I am getting the same error as bkibilko stated: 

"ERR (3): Warning: Redis::pipeline(): Already in pipeline mode in magento.../Credis/Client.php on line 931"
The offending code is:`
// Multi and pipeline return self for chaining

                if($this->isMulti) {
                    call_user_func_array(array($this->redisMulti, $name), $args);
                    return $this;
                }`
I have yet to find anything relating to Magento on this and not sure if it is Causing some of the magento
 errors I am getting. So since I have no regression to do how do I fix this? It is adding 80-100 MB of errors in my magento system.log every day and no one but "I" am using it yet. Just while writing this it had add another 11 MB of the same error lines.
Can anyone explain the concept of this error? Perhaps just to kill the error messages unitl there is a resolution?

**Comment by michael-grunder** [comment]
Prior to phpredis 3.1.3 if you called `pipeline` when already in pipeline mode it would just ignore the call.  3.1.3 changed this behavior and started issuing warnings.

```php
$r->pipeline(); // enter pipeline mode
$r->pipeline(); // In <= 3.1.2 call is ignored, in 3.1.3 it issues a warning
```

This should be fixed in the latest develop branch however.

**Comment by valdislav** [comment]
@michael-grunder 
So magento2 use old version of credis library "colinmollenhour/cache-backend-redis": "1.6", fixed from 1.7:
https://github.com/colinmollenhour/credis/commit/74b2b703da5c58dc07fb97e8954bc63280b469bf

**Comment by ewasser** [comment]
Please change this behavior and omit the warning.

The warning will break live magento shops without any warning for the maintainer. Upgrading a magento shop to remove the multiple `$r->pipeline()` calls can be extremly complicated.

**Comment by yatsukhnenko** [comment]
3.1.4RC1 with fix of this issue is up on PECL as beta (stable next month)

**Comment by yatsukhnenko** [comment]
Fixed in 3.1.4
